scripts/LVGLImage.py

```diff
@@ -872,9 +872,19 @@ def from_png(self,
     def _png_to_indexed(self, cf: ColorFormat, filename: str):
         # convert to palette mode
         auto_cf = cf is None
-        reader = png.Reader(
-            bytes=PngQuant(256 if auto_cf else cf.ncolors).convert(filename))
-        w, h, rows, _ = reader.read()
+        print(filename)
+        # read the image data to get the metadata
+        reader = png.Reader(filename=filename)
+        w, h, rows, metadata = reader.read()
+
+        # to preserve original palette data only convert the image if needed. For this
+        # check if image has a palette and the requested palette size equals the existing one 
+        if not 'palette' in metadata or not auto_cf and not len(metadata['palette']) ==  2 ** cf.bpp:
```

Comment:
Can we use != instead
```suggestion
        if not 'palette' in metadata or not auto_cf and len(metadata['palette']) !=  2 ** cf.bpp:
```